Source/Plugin.BLE.Android/Adapter.cs

```diff
@@ -184,6 +185,34 @@ public override List<IDevice> GetSystemConnectedOrPairedDevices(Guid[] services
             return connectedDevices.Union(bondedDevices, new DeviceComparer()).Select(d => new Device(this, d, null, 0)).Cast<IDevice>().ToList();
         }
 
+        protected override async Task<IDevice> ConnectNativeAsync(string friendlyName, Guid uuid, Func<IDevice, bool> deviceFilter, CancellationToken cancellationToken = default)
+        {
+            if (uuid == Guid.Empty)
+            {
+                var taskCompletionSource = new TaskCompletionSource<IDevice>();
+                EventHandler<DeviceEventArgs> handler = (sender, args) =>
```

Comment:
does it make sense to put this (sender, args) => in its own method?

Reply:
We could, but it would require us to 
1) Expose the TaskCompletionSource for the whole class
2) Expose the event handler for the whole class
None of them really makes sense in other contexts than in this method. So personally, I would prefer not to